libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -49,10 +49,37 @@ do {                                            \
                        OUTOFGEOFENCE)
 
 
+static uint8_t counter = 0;     // For Debuging
+
+
 AP_GPS_SBF::AP_GPS_SBF(AP_GPS &_gps, AP_GPS::GPS_State &_state,
-                       AP_HAL::UARTDriver *_port) :
+                       AP_HAL::UARTDriver *_port, uint8_t _asterx_setup) :
     AP_GPS_Backend(_gps, _state, _port)
 {
+    switch(_asterx_setup) {
+        case 0:
+            _asterx_type_is_singleantenna = true;
```

Comment:
Rather then maintain three booleans your current structure is probably better fit with an enum for what type of GPS it is, since you've implemented this in such a way that only one of these can be active at once.

Reply:
Yes that was my intention. Due to compatibility. Those are three different products with its own initialization. For example the AsteRx-m2 or the AsteRx-m2a are not able to generate the INSNavGeod message. If this message is included, the receiver will return message invalid and refuses to start the streams.

Reply:
Oh I get that, but since only one can be true the enum actually obviates the need for the booleans entirely.